frontend/javascripts/oxalis/view/right-menu/advanced_search_popover.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import { Icon, Input, Tooltip, Popover } from "antd";
+import * as React from "react";
+import _ from "lodash";
+import ButtonComponent from "oxalis/view/components/button_component";
+
+import Shortcut from "libs/shortcut_component";
+
+const InputGroup = Input.Group;
+
+type Props<S> = {
+  data: { +[number]: S } | Array<S>,
+  searchKey: $Keys<S>,
+  onSelect: S => void,
+  children: *,
+  provideShortcut?: boolean,
+};
+
+type State = {
+  isVisible: boolean,
+  searchQuery: string,
+};
+
+export default class AdvancedSearchPopover<S: Object> extends React.PureComponent<Props<S>, State> {
+  constructor() {
+    super();
+    this.state = {
+      isVisible: false,
+      searchQuery: "",
+    };
+    this.availableOptions = [];
+  }
+
+  availableOptions: { +[number]: S } | Array<S> = [];
+  currentPosition: number = 0;
+
+  selectNextOptionWithOffset = (offset: number) => {
+    const numberOfAvailableOptions = _.values(this.availableOptions).length;
+    if (numberOfAvailableOptions === 0) {
+      return;
+    }
+    this.currentPosition = (this.currentPosition + offset) % numberOfAvailableOptions;
+    if (this.currentPosition < 0) {
+      this.currentPosition = numberOfAvailableOptions + this.currentPosition;
+    }
+    this.props.onSelect(_.values(this.availableOptions)[this.currentPosition]);
```

Comment:
I think, `currentPosition` should be in the state instead of in an instance variable. A good rule of thumb on whether to put sth into the state vs instance variable, is whether it affects the rendering or not. As it does affect the rendering, changing it should cause a rerender of the component. Currently this works, nevertheless, but only because the `prop.onSelect` ultimately causes a prop change (because the activeNodeId/GroupId changed) which then causes a rerender.